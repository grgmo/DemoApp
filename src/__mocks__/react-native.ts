import RN from 'react-native';

RN.NativeModules.AppInfo = {
  getConstants: () => ({
    appVersion: '1.0.0',
  }),
};
module.exports = RN;
