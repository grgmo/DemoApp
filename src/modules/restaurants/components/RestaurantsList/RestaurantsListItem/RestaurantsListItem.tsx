import React, {FC, useCallback} from 'react';
import {Pressable, PressableStateCallbackType, StyleSheet} from 'react-native';
import theme from '../../../../../common/theme';
import AppText from '../../../../../common/components/AppText/AppText';

type RestaurantsListItemProps = {
  title: string;
  street: string;
  locality: string;
  postCode: string;
  url: string;
  onPress: (url: string) => void;
};

const RestaurantsListItem: FC<RestaurantsListItemProps> = ({
  title,
  street,
  locality,
  postCode,
  url,
  onPress,
}) => {
  const itemStyle = ({pressed}: PressableStateCallbackType) => [
    {
      backgroundColor: pressed ? theme.COLOR.whitesmoke : theme.COLOR.white,
    },
    styles.listItem,
  ];

  const handlePress = useCallback(() => {
    onPress(url);
  }, [onPress, url]);

  return (
    <Pressable
      accessibilityRole="button"
      style={itemStyle}
      onPress={handlePress}>
      <AppText type="h2" text={title} bold style={styles.title} />
      <AppText type="h3" text={street} />
      <AppText type="h3" text={locality} />
      <AppText type="h3" text={postCode} />
    </Pressable>
  );
};

const styles = StyleSheet.create({
  listItem: {
    marginHorizontal: theme.SPACE.SMALL,
    paddingHorizontal: theme.SPACE.SMALL,
    paddingVertical: theme.SPACE.MEDIUM,
    borderRadius: 2,
    borderWidth: 1.5,
    borderColor: theme.COLOR.lightGrey,
  },
  title: {
    marginBottom: theme.SPACE.SMALL,
  },
});

export default RestaurantsListItem;
