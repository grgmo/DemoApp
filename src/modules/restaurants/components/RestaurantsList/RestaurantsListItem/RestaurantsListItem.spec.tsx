import React from 'react';
import {fireEvent, render, RenderAPI} from '@testing-library/react-native';
import RestaurantsListItem from './RestaurantsListItem';

describe('RestaurantsListItem', () => {
  const mockItem = {
    title: 'title',
    street: 'street',
    locality: 'locality',
    postCode: 'postCode',
    url: 'url',
    onPress: jest.fn(),
  };
  let listItem: RenderAPI;

  beforeEach(() => {
    listItem = render(<RestaurantsListItem {...mockItem} />);
  });

  it('should render list item correctly', () => {
    expect(listItem.toJSON()).toMatchSnapshot();
  });

  it('should call onPress with correct url on item press', () => {
    const {getByRole} = listItem;

    fireEvent.press(getByRole('button'));

    expect(mockItem.onPress).toHaveBeenCalledTimes(1);
    expect(mockItem.onPress).toHaveBeenCalledWith(mockItem.url);
  });
});
