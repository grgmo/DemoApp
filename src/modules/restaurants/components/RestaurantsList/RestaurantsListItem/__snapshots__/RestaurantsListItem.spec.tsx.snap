// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantsListItem should render list item correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
      },
      Object {
        "borderColor": "lightgrey",
        "borderRadius": 2,
        "borderWidth": 1.5,
        "marginHorizontal": 10,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 14,
        },
        Object {
          "marginBottom": 10,
        },
        Object {
          "fontSize": 18,
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
    title
  </Text>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 14,
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
    street
  </Text>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 14,
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
    locality
  </Text>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 14,
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
    postCode
  </Text>
</View>
`;
